### ISSUE TITLE
unify power and window common parts

### ISSUE THREAD

**Issue by pmcdonal** [adematti/cosmopipe#5]
It would be good to unify, e.g., the catalog reading, cutting, etc. for window and power. E.g., here, after setting up zmin and zmax options for power, I am needing to duplicate them for window, even though the catalogs should be the same. 

https://github.com/adematti/cosmopipe/blob/0f15bd770b563dbaf33625afac54afc79297f654/cosmopipe/estimators/window_function/fft_window_function.py#L29

Catalogs probably should be read upstream and just passed down to power and window, and power should pass along things that are supposed to be the same like normalization.

**Comment by adematti** [comment]
Yes, I did hesitate to include an extra step of "catalog reading / preparation". The main reason I did not do that, is that in most cases this catalog preparation is trivial enough (typically computing weights) not to have to include an extra step. Also, this preparation step will mostly depend on whether we need only RA, DEC (angular correlation function), or positions, which is decided by the estimator. But we definitely can have that, yes, especially if we want less trivial changes in the survey selection.
